src/internal/context.go

```diff
@@ -78,3 +80,33 @@ func GetArtifactInfo(ctx context.Context) (art ArtifactInfo) {
 	}
 	return
 }
+
+// WithAuthMode returns a context with auth mode set
+func WithAuthMode(ctx context.Context, mode string) context.Context {
+	return setToContext(ctx, contextKeyAuthMode, mode)
+}
+
+// GetAuthMode gets the auth mode from the context
+func GetAuthMode(ctx context.Context) string {
```

Comment:
I just realized it will be slightly better if we add a type assertion in all `GetXXX` funcs in this file (not only the ones you added).

It would be better like 
```
mode, _ := value.(string)
```
such that there won't be panic.